pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -480,9 +502,13 @@ func (s *DiscoveryServer) initConnection(node *core.Node, con *XdsConnection) er
 	con.mu.Lock()
 	con.node = proxy
 	con.ConID = connectionID(node.Id)
+	con.meta = node
```

Comment:
What do you need from the node?

Reply:
It is sent on disconnect events, as well as with the 'nacks'.

As I mentioned - we can also duplicate and create new protos - but using the native Envoy types and not making copies and allocs would be my first choice. 

We are using XDS as a pubsub transport - so messages need to be self-contained, the 'nack' needs to have info about which node it applies to.